### PR TITLE
HPCC-28450 Rationalize filecopy transfer endpoint tracing

### PR DESCRIPTION
**Pull request by jakesmith**
Also reduce overly verbose tracing in numParallelConnections (was tracing per call, and there could be e.g. 400+ calls)

Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by AttilaVamos — APPROVED** [review]
The changes seems OK.
Tested locally on bare-metal system with 4 Thor slaves.
Perhaps I misunderstand something, but I can't see less trace/log entries generated by a spray operation.

**Comment by jakesmith** [comment]
> Perhaps I misunderstand something, but I can't see less trace/log entries generated by a spray operation.

You may need to run on cloud to notice.
In cloud, it called numParallelSlaves() for every partition, so spraying 400 ways, results in 400 log entries from it.

**Review by ghalliday — COMMENTED** [review]
@jakesmith 1 question

File: dali/ft/filecopy.cpp [hunk 1]
```diff
@@ -346,34 +348,15 @@ void FileTransferThread::prepareCmd(MemoryBuffer &msg, unsigned version)
...
-    StringBuffer url;
-    ep.getUrlStr(url);
-
     Owned<ISocket> socket;
     MemoryBuffer msg;
     msg.setEndian(__BIG_ENDIAN);
     if (sprayer.useFtSlave)
     {
         StringBuffer tmp;
+        SocketEndpoint connectEP("localhost");
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply dali/ft/filecopy.cpp]: whoops, yes. Now changed.

**Comment by jakesmith** [comment]
@ghalliday - please review changes

**Review by ghalliday — COMMENTED** [review]
@jakesmith looks good.  Please squash.  Also, should this target 8.10.x or master.  It doesn't seem to be critical, and there is a slight chance it will break something.

**Review by ghalliday — APPROVED** [review]

**Comment by jakesmith** [comment]
> @jakesmith looks good. Please squash. Also, should this target 8.10.x or master. It doesn't seem to be critical, and there is a slight chance it will break something.

@ghalliday - now squashed and retargeted to master.

**Pull request merged**